docs/user/metal/install_upi.md

```diff
@@ -0,0 +1,305 @@
+
+```console
+$ mkdir test-bare-metal
+$ tree test-bare-metal
+test-bare-metal
+
+0 directories, 0 files
+```
+
+Copy *your* `install-config` to the `INSTALL_DIR`. For example using the `test-bare-metal` as our `INSTALL_DIR`,
+
+```console
+$ cp <your-instal-config> test-bare-metal/install-config.yaml
+$ tree test-bare-metal
+test-bare-metal
+└── install-config.yaml
+
+0 directories, 1 file
+```
+
+NOTE: The filename for `install-config` in the `INSTALL_DIR` must be `install-config.yaml`
+
+### Invoking the installer to get Ignition configs
+
+Given that you have setup the `INSTALL_DIR` with the appropriate `install-config`, you can create the Ignition configs by using the `create ignition-configs` target. For example,
+
+```console
+$ openshift-install --dir test-bare-metal create ignition-configs
+INFO Consuming "Install Config" from target directory
+$ tree test-bare-metal
+test-bare-metal
+├── auth
+│   └── kubeconfig
+├── bootstrap.ign
+├── master.ign
+└── worker.ign
+
+1 directory, 4 files
+```
+
+## Getting OS related assets for machines
+
+TODO RHEL CoreOS does not have assets for bare metal.
+
+## Booting machines with RHCOS and Ignition configs
+
+### Required kernel parameters during PXE
+
+* `rd.neednet=1`: [CoreOS Installer][coreos-installer] needs internet access to fetch the OS image that needs to be installed on the machine.
+
+* CoreOS Installer [arguments][coreos-installer-args] are required to be configured to install RHCOS and setup the Ignition config file for that machine.
+
+## Watching your installation
+
+### Monitor for bootstrap-complete
+
+The administrators can use the `user-provided-infrastructure bootstrap-complete` target of the OpenShift Installer to monitor cluster bootstrapping. The command succeeds when it notices `bootstrap-complete` event from Kubernetes APIServer. This event is generated by the bootstrap machine after the Kubernetes APIServer has been bootstrapped on the control plane machines. For example,
+
+```console
+$ openshift-install --dir test-bare-metal user-provided-infrastructure bootstrap-complete
+INFO Waiting up to 30m0s for the Kubernetes API at https://api.test.example.com:6443...
+INFO API v1.12.4+c53f462 up
+INFO Waiting up to 30m0s for the bootstrap-complete event...
+```
+
+## Monitor for cluster completion
+
+The administrators can use the `user-provided-infrastructure finish` target of the OpenShift Installer to monitor cluster completion. The command succeeds when it notices that Cluster Version Operator has completed rolling out the OpenShift cluster from Kubernetes APIServer.
+
+```console
+$ openshift-install user-provided-infrastructure finish
+INFO Waiting up to 30m0s for the cluster to initialize...
+```
+
+## Example Bare-Metal UPI deployment
+
+Terraform [templates][upi-metal-example] provides and example of using OpenShift Installer to create an UPI Bare Metal OpenShift cluster on Packet.net
```

Comment:
> ... provides an ...

Reply:
and

> create a UPI Bare

Would probably suggest we switch `Bare Metal` to `bare-metal` everywhere.

Reply:
+1. Use "bare-metal" as an adjective.